subsys/suit/platform/sdfw/src/suit_plat_ipuc.c

```diff
@@ -0,0 +1,199 @@
+		/* The only supported component type is MEM
+		 */
+		return NULL;
+	}
+
+	return component_id;
+}
+
+int suit_plat_ipuc_revoke(suit_component_t handle)
+{
+	struct zcbor_string *component_id = ipuc_component_id_from_handle(handle);
+
+	if (component_id == NULL) {
+		return SUIT_ERR_UNSUPPORTED_COMPONENT_ID;
+	}
+
+	k_mutex_lock(&ipuc_mutex, K_FOREVER);
+	for (size_t i = 0; i < CONFIG_SUIT_IPUC_SIZE; i++) {
+		suit_plat_ipuc_entry_t *p_entry = &components[i];
+
+		if (suit_compare_zcbor_strings(component_id, &p_entry->component_id)) {
+
+#ifdef CONFIG_SUIT_LOG_LEVEL_INF
+			intptr_t slot_address = 0;
+			size_t slot_size = 0;
+
+			if (suit_plat_decode_address_size(&p_entry->component_id, &slot_address,
+							  &slot_size) == SUIT_PLAT_SUCCESS) {
+				LOG_INF("Revoking IPUC, address: %p, size: %d bytes",
+					(void *)slot_address, slot_size);
+			}
+#endif
+			p_entry->component_id.value = NULL;
+			p_entry->component_id.len = 0;
+			p_entry->write_offset = 0;
+			break;
+		}
+	}
+
+	k_mutex_unlock(&ipuc_mutex);
+
+	return SUIT_SUCCESS;
+}
+
+int suit_plat_ipuc_declare(suit_component_t handle)
+{
+	suit_plat_ipuc_entry_t *allocated_entry = NULL;
+	struct zcbor_string *component_id = ipuc_component_id_from_handle(handle);
+
+	if (component_id == NULL) {
+		return SUIT_ERR_UNSUPPORTED_COMPONENT_ID;
+	}
+
+	k_mutex_lock(&ipuc_mutex, K_FOREVER);
+	for (size_t i = 0; i < CONFIG_SUIT_IPUC_SIZE; i++) {
+		suit_plat_ipuc_entry_t *p_entry = &components[i];
+
+		if (allocated_entry == NULL && p_entry->component_id.value == NULL) {
+			allocated_entry = p_entry;
+
+		} else if (suit_compare_zcbor_strings(component_id, &p_entry->component_id)) {
+			allocated_entry = p_entry;
+			break;
+		}
+	}
+
+	if (allocated_entry == NULL) {
+		k_mutex_unlock(&ipuc_mutex);
+		return SUIT_ERR_OVERFLOW;
+	}
+
+#ifdef CONFIG_SUIT_LOG_LEVEL_INF
+	intptr_t slot_address = 0;
+	size_t slot_size = 0;
+
+	if (suit_plat_decode_address_size(component_id, &slot_address, &slot_size) ==
+	    SUIT_PLAT_SUCCESS) {
+		LOG_INF("Declaring IPUC, address: %p, size: %d bytes", (void *)slot_address,
+			slot_size);
+	}
+#endif
+
+	allocated_entry->component_id.value = component_id->value;
+	allocated_entry->component_id.len = component_id->len;
+	allocated_entry->write_offset = 0;
+
+	k_mutex_unlock(&ipuc_mutex);
+	return SUIT_SUCCESS;
+}
+
+struct zcbor_string *suit_plat_find_sdfw_mirror_ipuc(size_t min_size)
+{
+	struct zcbor_string *component_id = NULL;
+
+	k_mutex_lock(&ipuc_mutex, K_FOREVER);
+	for (size_t i = 0; i < CONFIG_SUIT_IPUC_SIZE; i++) {
+
+		component_id = &components[i].component_id;
+
+		if (component_id->value != NULL) {
+
+			suit_component_type_t component_type = SUIT_COMPONENT_TYPE_UNSUPPORTED;
+
+			intptr_t slot_address = 0;
+			size_t slot_size = 0;
+
+			if (suit_plat_decode_component_type(component_id, &component_type) !=
+				    SUIT_PLAT_SUCCESS ||
+			    component_type != SUIT_COMPONENT_TYPE_MEM) {
+				continue;
```

Comment:
Looking at `suit_plat_ipuc_declare` (more specifically at `ipuc_component_id_from_handle`) a component of type different then `MEM` should never land in the `components` array. Shouldn't we return an error here?

Reply:
No, IPUC will have two use cases:

1. using its space as 'sdfw mirror', i.e. when SDFW candidate is placed in external flash
2. updating inactive component from appliaction.

1st case indeed can operate on MEM component only, but 2nd case may be supported on other component types as well